File: cluster-check-go/security-context.go
package main

import (
	"context"
	"fmt"
	v1 "github.com/openshift/client-go/security/clientset/versioned/typed/security/v1"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
)

func validateSCCs(securityClient v1.SecurityV1Interface, configCheck *Config) bool {
	var passed = true

	fmt.Println(Separator)
	fmt.Println("Checking security context constraints")
	fmt.Println(Separator)
	for _, scc := range configCheck.SecurityContexts {
		status := ""
		color := ""
		_, err := securityClient.SecurityContextConstraints().Get(context.TODO(), scc, metav1.GetOptions{})
		if err != nil {
			status = "Not Found"
			color = Red
			passed = false
			checkError(err)
		} else {
			status = "Found"
			color = Green
		}
		fmt.Printf("%s - %s%s %s\n", scc, color, status, Reset)
	}

	fmt.Print("\n")
	return passed
}

File: cluster-check-py/securityContext.py
import openshift as oc
import utils

from myTypes import Config
from kubernetes.client.rest import ApiException
from kubernetes.dynamic import DynamicClient


def validate_sccs(dyn_client,config_check: Config):
    print("---------------------------------------------------------")
    print("Checking security context constraints")
    print("---------------------------------------------------------")
    passed = True
    scc_client = dyn_client.resources.get(api_version='security.openshift.io/v1', kind='SecurityContextConstraints')
    for scc in config_check.SecurityContexts:
        status = ""
        color = ""
        try:
            temp= scc_client.get(name=scc)
            status = "Found"
            color = utils.Color.GREEN.value
        except ApiException as e:
            status = "Not Found"
            color = utils.Color.RED.value
            passed = False
            utils.check_error(e)
        print(f"{scc} - {color}{status}\033[0m")
    print()
    return passed